test/unit/coroutine/await.cpp

```diff
@@ -0,0 +1,115 @@
+#include <util/time.hpp>
+
+#include <yaclib/async/run.hpp>
+#include <yaclib/coroutine/await.hpp>
+#include <yaclib/coroutine/future_coro_traits.hpp>
+#include <yaclib/executor/thread_pool.hpp>
+#include <yaclib/fault/thread.hpp>
+
+#include <array>
+#include <exception>
+#include <iostream>
+#include <stack>
+#include <utility>
+
+#include <gtest/gtest.h>
+
+namespace {
+using namespace std::chrono_literals;
+
+TEST(Await, JustWorksPack) {
+  auto tp = yaclib::MakeThreadPool();
+  auto coro = [&](yaclib::IThreadPoolPtr tp) -> yaclib::Future<int> {
+    auto f1 = yaclib::Run(tp, [] {
+      yaclib_std::this_thread::sleep_for(10ms);
+      return 1;
+    });
+    auto f2 = yaclib::Run(tp, [] {
+      return 2;
+    });
+
+    co_await Await(f1, f2);
+    co_return std::move(f1).Get().Ok() + std::move(f2).Get().Ok();
+  };
+  auto f = coro(tp);
+  EXPECT_EQ(std::move(f).Get().Ok(), 3);
+  tp->HardStop();
+  tp->Wait();
+}
+
+TEST(Await, JusWorksRange) {
+  auto tp = yaclib::MakeThreadPool();
+  auto coro = [&](yaclib::IThreadPoolPtr tp) -> yaclib::Future<int> {
+    std::array<yaclib::Future<int>, 2> arr;
+    arr[0] = yaclib::Run(tp, [] {
+      yaclib_std::this_thread::sleep_for(500ms);
+      return 1;
+    });
+    arr[1] = yaclib::Run(tp, [] {
+      return 2;
+    });
+
+    co_await yaclib::Await(arr.begin(), arr.end());
+    co_return std::move(arr[0]).GetUnsafe().Ok() + std::move(arr[1]).GetUnsafe().Ok();
+  };
+  auto future = coro(tp);
+  EXPECT_EQ(std::move(future).Get().Ok(), 3);
+  tp->HardStop();
+  tp->Wait();
+}
+
+TEST(Await, CheckSuspend) {
+  int counter = 0;
+  auto tp = yaclib::MakeThreadPool();
+
+  auto coro = [&]() -> yaclib::Future<void> {
+    counter = 1;
+    auto future1 = yaclib::Run(tp, [] {
+      yaclib_std::this_thread::sleep_for(500ms);
+    });
+    auto future2 = yaclib::Run(tp, [] {
+      yaclib_std::this_thread::sleep_for(1000ms);
+    });
+
+    co_await Await(future1, future2);
+    counter = 2;
+    co_return;
+  };
+
+  auto outer_future = coro();
+  EXPECT_EQ(1, counter);
+  std::move(outer_future).Get();
+  EXPECT_EQ(2, counter);
```

Comment:
please add a time check, also you can use smaller than 500 and 1000ms time but you should use `YACLIB_CI_SLOWDOWN`